components/src/components/shared/sidebar/Sidebar.tsx

```diff
@@ -50,15 +50,16 @@ interface SidebarProps {
   checkoutStages?: CheckoutStage[];
   error?: string;
   isLoading: boolean;
+  isPaymentMethodRequired: boolean;
   paymentMethodId?: string;
   promoCode?: string | null;
-  requiresPayment: boolean;
   setCheckoutStage?: (stage: string) => void;
   setError: (msg?: string) => void;
   setIsLoading: Dispatch<SetStateAction<boolean>>;
   updatePromoCode?: (code: string | null) => void;
   showHeader?: boolean;
-  willTrial?: boolean;
+  shouldTrial?: boolean;
+  willTrialFree?: boolean;
```

Comment:
My comment is the same for this props object. Let's rename it to `willTrialWithoutPaymentMethod`